aptos-move/framework/aptos-framework/sources/native_bridge.move

```diff
@@ -560,6 +649,21 @@ module aptos_framework::native_bridge {
         assert!(borrow_global<BridgeConfig>(@aptos_framework).bridge_relayer == signer::address_of(caller), EINVALID_BRIDGE_RELAYER);
     }
 
+    /// Asserts that the rate limit budget is not exceeded.
+    /// 
+    /// @param amount The amount to be transferred.
+    fun assert_rate_limit_budget_not_exceeded(amount: u64) acquires SmartTableWrapper, BridgeConfig {
+        let insurance_fund = borrow_global<BridgeConfig>(@aptos_framework).insurance_fund;
+        let risk_denominator = borrow_global<BridgeConfig>(@aptos_framework).risk_denominator;
+        let table = borrow_global_mut<SmartTableWrapper<u64, u64>>(@aptos_framework);
+        
+        let day = timestamp::now_seconds() / 86400;
+        let current_budget = smart_table::borrow_mut_with_default(&mut table.inner, day, 0);
+        smart_table::upsert(&mut table.inner, day, *current_budget + amount);
+        let rate_limit = coin::balance<AptosCoin>(insurance_fund) / risk_denominator;
+        assert!(*smart_table::borrow(&table.inner, day) < rate_limit, ERATE_LIMIT_EXCEEDED);
```

Comment:
There seems to be an error in the units

the left value is a budget unit, whereas rate_limit is a budget/time-unit (here max_budget/day) .. 

for example one_day_budget = rate_limit * 1 (days)  

however the budget can be consumed within a risk_period, which could be 8 hours, 24hours, 48 hours, etc.

maybe we should approach this different:
- window = risk_denominator ( = reaction time )
- budget_for_window = insurance_fund
- assert ( current_budget + amount < budget_for_window )

Reply:
why would we want to get this correct? Because the window length of budgets probably will change. specifically the Governance Operator may set the window_size to 8 hours, which means in a first approach (without moving window) the max_budget will reset after the window time.